src/one/nio/http/HttpSession.java

```diff
@@ -134,15 +134,22 @@ protected int processHttpBuffer(byte[] buffer, int length) throws IOException, H
     }
 
     protected Request parseRequest(byte[] buffer, int start, int length) throws HttpException {
-        boolean http11 = length > 13 && buffer[start + length - 1] == '1';
-        if (length > 13 && Utf8.startsWith(Request.VERB_GET, buffer, start)) {
-            return new Request(Request.METHOD_GET, Utf8.read(buffer, start + 4, length - 13), http11);
-        } else if (length > 14 && Utf8.startsWith(Request.VERB_POST, buffer, start)) {
-            return new Request(Request.METHOD_POST, Utf8.read(buffer, start + 5, length - 14), http11);
-        } else if (length > 14 && Utf8.startsWith(Request.VERB_HEAD, buffer, start)) {
-            return new Request(Request.METHOD_HEAD, Utf8.read(buffer, start + 5, length - 14), http11);
-        } else if (length > 17 && Utf8.startsWith(Request.VERB_OPTIONS, buffer, start)) {
-            return new Request(Request.METHOD_OPTIONS, Utf8.read(buffer, start + 8, length - 17), http11);
+        // <VERB> <PATH> HTTP/1.{0|1}
+        for (int i = 1; i < Request.VERBS.length; i++) {
+            // "<VERB> "
+            final byte[] verb = Request.VERBS[i];
+            if (length > verb.length && Utf8.startsWith(verb, buffer, start)) {
+                return new Request(
+                        i,
+                        Utf8.read(
+                                buffer,
+                                // Strip "<VERB> "
+                                start + verb.length,
+                                // Trim " HTTP/1.{0,1}"
+                                length - (verb.length + 9)),
```

Comment:
Won't it be ArrayIndexOutBoundsException for a malformed request like "GET \n"

Reply:
Yes, you are right. Added a unit test and a fix.